include/dgl/graph_serializer.h

```diff
@@ -0,0 +1,26 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file graph/graph_serializer.cc
+ * \brief DGL serializer APIs
+ */
+
+#ifndef DGL_GRAPH_SERIALIZER_H_
+#define DGL_GRAPH_SERIALIZER_H_
+
+#include <memory>
+namespace dgl {
+class Serializer {
```

Comment:
Do you it will be easier if the default constructor of any serializable object is public?

Reply:
Yes. Should I do so?

Reply:
I think your solution is also fine, but please add more comments for this class such as code snippet of declaring it as a friend class, etc.